src/login/logind.h

```diff
@@ -96,6 +96,9 @@ struct Manager {
         HandleAction idle_action;
         bool was_idle;
 
+        sd_event_source *stop_idle_session_event_source;
```

Comment:
I don't understand why this is a single event source on the Manager object? There should be one per Session object instead, stored in that object...

Reply:
Hmm, I thought that because this is a global option for all sessions we don't really need per session evet source. I mean if we would later want to allow specifing different values for each session then we could move this down to session objects.

Reply:
this really should be in the sessions. The whole reason we have sd_event_add_time() is so that we can efficiently implement a prioq of pending time events and run them all over the very same timerfd handle, which becomes very handy here. Basically you implemented a prioq yourself here in the original patch even though the one sd_event_add_time() is much nicer because more widely used.